src/variant.ts

```diff
@@ -0,0 +1,93 @@
+import { Context } from './context';
+import { FeatureInterface } from './feature';
+import * as murmurHash3 from 'murmurhash3js';
+
+enum PayloadType {
+    STRING = 'string',
+}
+
+export interface PayloadInterface {
+    type: PayloadType;
+    value: string;
+}
+
+export interface VariantInterface {
+    name: string;
+    weight: number;
+    payload: PayloadInterface;
+}
+
+export interface VariantResult {
+    name: string;
+    payload: PayloadInterface;
+    enabled: boolean;
+}
+
+export function getDefaultVariantResult(): VariantResult {
+    return {
+        name: 'disabled',
+        enabled: false,
+        payload: getDefaultPayload(),
+    };
+}
+
+export function getDefaultPayload(): PayloadInterface {
+    return {
+        type: PayloadType.STRING,
+        value: '',
+    };
+}
+
+const stickynessSelectors = ['userId', 'sessionId', 'remoteAddress'];
+function getUserFromContext(context: Context): string {
+    let result = '';
+    stickynessSelectors.some(
+        (key: string): boolean => {
+            const value = context[key];
+            if (typeof value === 'string' && value !== '') {
+                result = value;
+                return true;
+            }
+            return false;
+        },
+    );
+    return result;
+}
+
+export function selectVariant(
+    feature: FeatureInterface,
+    context: Context,
+): VariantInterface | null {
+    const userIdString = getUserFromContext(context);
+
+    if (!userIdString) {
+        return null;
```

Comment:
We need to return a Variant even if there is no userId on the context. 

If the context is missing at all we probably need to just do some random things, this needs to be documented in the spec somehow.

Reply:
done